libopenage/renderer/camera/camera.cpp

```diff
@@ -102,15 +102,37 @@ void Camera::look_at_coord(coord::scene3 coord_pos) {
 	this->look_at_scene(scene_pos);
 }
 
-void Camera::move_to(Eigen::Vector3f scene_pos) {
-	// TODO: Check and set bounds for where the camera can go and check them here
+void Camera::move_to(Eigen::Vector3f scene_pos,
+                     std::optional<std::pair<float, float>> x_bounds,
+                     std::optional<std::pair<float, float>> z_bounds) {
```

Comment:
Instead of using two pairs for the bounds, it probably makes sense to make these parameters a struct and pass that.